### PR TITLE
chore: tag clickhouse queries made from decide/capture endpoints

### PR DESCRIPTION
**Pull request by macobo**
We had an incident where clickhouse indexes/projections seemed to affect ingestion. That's weird - let's find out what queries there are being made.

Ideally we'd also tag the pod but I don't believe we have access to that info in the app. Correct if I'm wrong.

### CODE REVIEW

File: posthog/middleware.py [hunk 1]
```diff
@@ -223,6 +223,11 @@ def __init__(self, get_response):
 
     def __call__(self, request: HttpRequest):
         if request.path == "/decide/" or request.path == "/decide":
```
Comment (macobo) [inline posthog/middleware.py:225]: Q: Why do we short circuit decide but not capture? cc @xvello you have been working with these middlewares a lot - is this the right place to be even tagging things?

Comment (xvello) [reply posthog/middleware.py:225]: Heya. I have [a PR open](https://github.com/PostHog/posthog/pull/13319) to also shortcut on capture, but I suspended it due to (relevant) concerns about losing the observability.

Today showed us that we do lack observability on what is keeping the gunicorn workers busy.

**Comment by xvello** [comment]
What is the perf impact of that change? Would it be helpful & cheap to have on all queries instead of only decide?

> Ideally we'd also tag the pod 

The `HOSTNAME` envvar is set to the pod name (or the container name in docker-compose stacks), so we could source it from there. It is also possible to get it into a specific envvar via [the downwards API](https://kubernetes.io/docs/tasks/inject-data-application/environment-variable-expose-pod-information/) if we want to.

**Comment by macobo** [comment]
Added CONTAINER_HOSTNAME setting as a tag to all queries.

> What is the perf impact of that change? 

zero. Single update to a thread-local dictionary.

> Would it be helpful & cheap to have on all queries instead of only decide?

It's already on all requests _but_ decide (and healthcheck).

**Review by xvello — APPROVED** [review]

**Comment by xvello** [comment]
>> Would it be helpful & cheap to have on all queries instead of only decide?
> It's already on all requests but decide (and healthcheck).

Thanks. So we're doing a lightweigth version of the `CHQueries` middleware?

**Comment by macobo** [comment]
Basically yes. CHMiddleware also tries to tag things with user/team id which we don't have access to here.

**Pull request merged**